src/services/redisUserMetadataStorage.ts

```diff
@@ -23,14 +26,33 @@ export default class RedisUserMetadataStorage extends RedisStorageUtils
 
   /**
    * {@inheritDoc}
-   *
-   * This method doesn't support atomic operations on concurrency scenario.
-   * Story https://www.pivotaltracker.com/story/show/167064659
+   * This method uses Optimistic Lock to prevent race condition
+   * during write operations of user metadata
+   * @see https://github.com/NodeRedis/node_redis#optimistic-locks
    */
   public async set(
     user: User,
     payload: UserMetadata
   ): Promise<Either<Error, boolean>> {
+    // In order to work properly, optimistic lock needs to be initialized on different
+    // redis client instances
```

Comment:
lgtm, please add a link to https://github.com/NodeRedis/node_redis/issues/1320#issuecomment-373200541

Reply:
Fixed in 809bc57